lib/client/interfaces.go

```diff
@@ -288,3 +308,34 @@ func (k *Key) CheckCert() error {
 
 	return nil
 }
+
+// HostKeyCallback returns an ssh.HostKeyCallback that validates host
+// keys/certs against SSH CAs in the Key.
+func (k *Key) HostKeyCallback() (ssh.HostKeyCallback, error) {
+	var trustedKeys []ssh.PublicKey
+	for _, caCert := range k.SSHCAs() {
+		_, _, publicKey, _, _, err := ssh.ParseKnownHosts(caCert)
+		if err != nil {
+			return nil, trace.BadParameter("failed parsing CA cert: %v; raw CA cert line: %q", err, caCert)
+		}
+		trustedKeys = append(trustedKeys, publicKey)
+	}
+	// No CAs are provided, return a nil callback which will prompt the user
+	// for trust.
+	if len(trustedKeys) == 0 {
+		return nil, nil
+	}
```

Comment:
Unexpected behavior for a public API.  Either document this behavior very explicitly in function doc comments or return a `NotFoundError` and then handle it explicitly at the points where a `nil` value for `HostKeyCallback` is OK.